shared/engine/index.js

```diff
@@ -173,13 +176,27 @@ class Engine {
 
   _wrapResponseOnceOnly (method, param, response, waitingHandler) {
     const {sessionID} = param
+    const {seqid} = response || {seqid: 0}
 
     let once = false
+
+    const cleanup = () => {
+      delete this.sessionIDToResponse[sessionID]
+      delete this.seqIDToWrappedResponse[seqid]
+      if (waitingHandler) {
+        waitingHandler(true, method, sessionID)
```

Comment:
? this is how it works. cleanup is just me moving that code up into a function. both places call true